### ISSUE TITLE
Test status doesn't update if test file is deleted

### ISSUE THREAD

**Issue by dar5hak** [vitest-dev/vitest#1005]
### Describe the bug

If I run Vitest in watch mode, it reports the number of tests passing or failing as expected. But if I delete a test file, the said numbers don't update.

### Reproduction

Have two test files, say `a.test.js` and `b.test.js`. Each of them have one failing test, so we see 2 suites failing, 2 tests failing.

Now, we delete `b.test.js`. The status stays the same: 2 suites failing, 2 tests failing.

After quitting and restarting Vitest, it reports the correct count.

### System Info

```shell
System:
    OS: macOS 12.3
    CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
    Memory: 143.12 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.14.0 - ~/.volta/tools/image/node/16.14.0/bin/node
    npm: 8.3.1 - ~/.volta/tools/image/node/16.14.0/bin/npm
    Watchman: 2022.03.14.00 - /usr/local/bin/watchman
  Browsers:
    Firefox: 98.0.1
    Safari: 15.4
  npmPackages:
    @vitejs/plugin-react: ^1.0.7 => 1.2.0
    vite: ^2.8.0 => 2.8.6
    vitest: ^0.7.0 => 0.7.0
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by Xwil** [comment]
Did you modify existing test cases after deleting `b.test.ts`?

I've tried to reproduce this problem as you describe.

1. Create test file `a.test.ts`
```ts
import { describe, expect, it } from 'vitest';

describe('suit A', () => {
  it('case A1', () => {
    expect(0).toBe(1);
  });
});

```
2. Create test file `b.test.ts`
```ts
import { describe, expect, it } from 'vitest';

describe('suit B', () => {
  it('case B1', () => {
    expect(0).toBe(1);
  });
});

```

3. Run vitest

Now I got 2 files and 2 tests failed.

4. Delete `b.test.ts` 

I still got 2 files and 2 tests failed.

Then modify `a.test.ts` or add new cases, vitest refreshed.

Maybe vitest does not watch files change, just collect the result of test cases. Delete file does not trigger rerunning.

So, maybe it's not a bug?
